### ISSUE TITLE
Just updated source tree, build fails.

### ISSUE THREAD

**Issue by phalexo** [pytorch/pytorch#8447]
[ 28%] Building C object confu-deps/NNPACK/CMakeFiles/nnpack.dir/src/softmax-output.c.o
[ 28%] Building C object confu-deps/NNPACK/CMakeFiles/nnpack.dir/src/x86_64-fma/softmax.c.o
[ 28%] Building C object confu-deps/NNPACK/CMakeFiles/nnpack.dir/src/convolution-output.c.o
[ 28%] Building C object confu-deps/NNPACK/CMakeFiles/nnpack.dir/src/relu-input-gradient.c.o
[ 28%] Linking C static library ../../lib/libnnpack.a
[ 28%] Built target nnpack
make: *** [all] Error 2
Failed to run 'bash tools/build_pytorch_libs.sh --with-cuda --with-nnpack --with-mkldnn nccl caffe2 nanopb libshm gloo THD c10d'
(pytorch) developer@theano:~/pytorch$

**Comment by soumith** [comment]
could we see a full log?

**Comment by phalexo** [comment]
[log.txt](https://github.com/pytorch/pytorch/files/2100377/log.txt)

**Comment by phalexo** [comment]
@soumith  The entire output to the error is attached in the previous post.

**Comment by phalexo** [comment]
cc1plus: error: unrecognized command line option '-Wno-gnu-zero-variadic-macro-arguments' [-Werror]
cc1plus: error: unrecognized command line option '-Wno-aligned-allocation-unavailable' [-Werror]
cc1plus: error: unrecognized command line option '-Wno-inconsistent-missing-override' [-Werror]
cc1plus: error: unrecognized command line option '-Wno-unused-private-field' [-Werror]
cc1plus: error: unrecognized command line option '-Wno-unknown-warning-option' [-Werror]
cc1plus: error: unrecognized command line option '-Wno-typedef-redefinition' [-Werror]
cc1plus: error: unrecognized command line option '-Wno-invalid-partial-specialization' [-Werror]
cc1plus: all warnings being treated as errors
Scanning dependencies of target tbb_static
make[2]: *** [c10/CMakeFiles/c10.dir/dummy.cpp.o] Error 1
make[1]: *** [c10/CMakeFiles/c10.dir/all] Error 2
@soumith I have updated the gcc, and g++ to version 5 but that did not fix the problem.

**Comment by phalexo** [comment]
@soumith I was able to force cmake to build it by suppressing warnings with -w passed to the compiler.
Not a very good solution. So, some warnings now get promoted to an error and cause this failure.

**Comment by ezyang** [comment]
CC @goldsborough

**Comment by vadimkantorov** [comment]
Also hitting a bunch of warnings:
```
cc1: warning: command line option ‘-fvisibility-inlines-hidden’ is valid for C++/ObjC++ but not for C
cc1: warning: unrecognized command line option "-Wno-aligned-allocation-unavailable"
cc1: warning: unrecognized command line option "-Wno-inconsistent-missing-override"
cc1: warning: unrecognized command line option "-Wno-unused-private-field"
cc1: warning: unrecognized command line option "-Wno-unknown-warning-option"
cc1: warning: unrecognized command line option "-Wno-typedef-redefinition"
cc1: warning: unrecognized command line option "-Wno-invalid-partial-specialization"
```

**Comment by zou3519** [comment]
cc @goldsborough

**Comment by goldsborough** [comment]
I don't really have anything to do with this. This started with the recent build system changes. Maybe @orionr can look into this. I also observe in my builds that these warnings are emitted, but the builds proceed fine.

**Comment by goldsborough** [comment]
I think we used to pass these warning flags only for clang, but once we removed all the `target_compile_options` and instead use `CMAKE_CXX_FLAGS` (which is really bad), all this nuanced configuration got scrapped

**Comment by goldsborough** [comment]
From what I can see `no-aligned-allocation-unavailable` is a clang specific warning, so the above is probably the case. @orionr maybe we need add certain flags to `CMAKE_CXX_FLAGS` only when the compiler is clang?

**Comment by phalexo** [comment]
To force a build, stick "-w" as the first flag to be passed to the C
compiler in the CMake***.txt file.

**Comment by dami23** [comment]
@phalexo have you solved the problem "cc1plus: error: unrecognized command line ...". I am facing the same situation, could you give me a cue to solve it? Thank you very much!!

**Comment by phalexo** [comment]
I already said 4 days ago, that you can force PyTorch to be built by
suppressing all warnings with "-w" It is not a solution, it is a
work-around.
Modify CMake*.txt, find where all the flags are being passed to the C
compiler and put a new line in front with the "-w" flag.

**Comment by phalexo** [comment]
# ---[ Build flags
set(CMAKE_C_STANDARD 99)
set(CMAKE_CXX_STANDARD 11)
if(NOT MSVC)
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O2 -fPIC")
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-narrowing")
  # Eigen fails to build with some versions, so convert this to a warning
  # Details at http://eigen.tuxfamily.org/bz/show_bug.cgi?id=1459
  # alexo added -w flag
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -w")
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}
-Wno-invalid-partial-specialization")
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wextra")
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-missing-field-initializers")
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-type-limits")
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-typedef-redefinition")
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unknown-warning-option")
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unknown-pragmas")
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-sign-compare")
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-parameter")
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-variable")
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-function")
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-private-field")
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-result")
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}
-Wno-inconsistent-missing-override")
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}
-Wno-aligned-allocation-unavailable")
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}
-Wno-error=deprecated-declarations")
